### ISSUE TITLE
Cannot initialize session after long time idle or after hibernation of system

### ISSUE THREAD

**Issue by gmc-holle** [rbuehlma/pvr.zattoo#102]
Hi,

first of all thanks for this great add-on. It is working very well except when Kodi is idling too long, e.g. stopping live tv because of going to sleep ;), or after the system comes back from hibernation (system was suspended and resumed). Then the add-on gets a 403 from zattoo.com and tries to re-initialize the session resulting in no TV anymore. One solution - of course - is to reboot the system or to disable and enable the add-on (my preferred way). Both solution have the disadvantage that all EPG information are lost for some time :(

At least hibernation worked long time ago. If remember right hibernation and re-initializing session worked till beginning of this year.

Is there anything I can do? Any setting to tweak?

Here is the part of the log showing the problem.

```2020-09-23 20:49:09.905 T:139745537195648   ERROR: CCurlFile::FillBuffer - Failed: HTTP returned error 403
2020-09-23 20:49:09.905 T:139745537195648   ERROR: CCurlFile::Open failed with code 403 for https://zattoo.com/zapi/watch/live/vox_up_de:
2020-09-23 20:49:09.906 T:139745537195648   ERROR: AddOnLog: Zattoo PVR Client: Open URL failed. Try to re-init session.
2020-09-23 20:49:09.996 T:139745537195648   ERROR: CCurlFile::FillBuffer - Failed: HTTP returned error 403
2020-09-23 20:49:09.996 T:139745537195648   ERROR: CCurlFile::Open failed with code 403 for https://zattoo.com/zapi/v2/account/login:
2020-09-23 20:49:09.996 T:139745537195648   ERROR: AddOnLog: Zattoo PVR Client: Login failed.
2020-09-23 20:49:09.996 T:139745537195648   ERROR: AddOnLog: Zattoo PVR Client: Re-init of session. Failed.
2020-09-23 20:49:09.996 T:139745537195648   ERROR: FillChannelStreamFileItem: Add-on 'Zattoo PVR Add-on:connected' returned an error: the command failed```

Regards,
Stephan

**Comment by rbuehlma** [comment]
Please provide a complete debug log. Your snipped does not contain enough information.

**Comment by rbuehlma** [comment]
Closing due to no feedback. Feel free to re-open the ticket with additional information.

**Comment by gmc-holle** [comment]
Hi,

sometimes it just works ... till last weekend. So I can now supply a debug log starting from the time we stopped looking a movie via Amazon VOD addon, took a break of 15 minutes and wanted to look "kika" station. I hope the log shows that it did not work because of HTTP status code 403 (Forbidden) and also re-initializing the session did not work. At the end of the log you will see that deactivating and reactivating the Zattoo PVR addon did the trick (again).

Regards
Stephan

Debug Log: [kodi_pvr_zattoo_403.log](https://github.com/rbuehlma/pvr.zattoo/files/5369285/kodi_pvr_zattoo_403.log)

**Comment by rbuehlma** [comment]
This should hopefully be fixed in 18.1.21
